src/App.tsx

```diff
@@ -1,24 +1,207 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import {
+  deleteTodo,
+  getTodos,
+  postTodo,
+  patchTodo,
+} from './api/todos';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import { ErrorMessage } from './components/ErrorMessage';
+import { AppError } from './types/AppError';
+import { filterTodos, FilterType } from './helpers/filterTodos';
+import { TodoItem } from './components/TodoItem';
+import { getActiveTodosCount, getHasCompletedTodos } from './helpers/getters';
 
-const USER_ID = 0;
+const USER_ID = 6950;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState(FilterType.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errorType, setErrorType] = useState(AppError.None);
+  const [hasError, setHasError] = useState(false);
+  const [loadingTodosIds, setLoadingTodosIds] = useState([0]);
+
```

Comment:
It's better to initialize 'loadingTodosIds' with an empty array instead of containing a 0, like 'useState<number[]>([])'